atmat/atphysics/nafflib/calcnaff.m

```diff
@@ -71,15 +65,6 @@
     frequency = 0; amplitude = 0;  phase = 0;
 else % Frequency map analysis
     [frequency,amplitude,phase] = nafflib(Y, Yp, WindowType,nfreq,DebugFlag);
-    %It seems there is a bug in nafflib, something returns nan even for valid data 
-    niter = 0;
-    while any(isnan(frequency)) && (niter < nitermax)
-        pause(2);
-        fprintf('Warning Nan returned by NAFF (x%d)\n', niter);
-        niter = niter +1;
-        [frequency,amplitude,phase] = nafflib(Y, Yp, WindowType,nfreq,1); % add debugging
-    end
```

Comment:
So apparently, up to now, if the output is wrong, we try again until it gets right !
This looks crazy, your fix is really welcome !

Reply:
Line #7 should become
 2. Yp - angle vector

Line 36 should become
NT = 9996; % divisible by 6

line #60 
elseif any(isnan(Y(1,:)))
should read
elseif any(isnan(Yp(1,:)))